ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -339,6 +339,31 @@ module CBuf = struct
     let read = Unix.read fd x.buffer next len in
     if read = 0 then x.r_closed <- true ;
     x.len <- x.len + read
+
+  (* Read data from the fd to the buffer and return the newly read data *)
```

Comment:
Is this in some way re-implementing buffered IO as it is available from `in_chan`? Unbuffered IO (`Unix.read`) already reads what is available into a buffer and returns the number of bytes it has read. What problem is this code solving?